### PR TITLE
ldc: 1.20.1 -> 1.24.0

### PR DESCRIPTION
**Pull request by fabianhjr**
#### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lionello** [comment]
Fails here on Darwin for me: 
```
llvm-config: unknown component name: windowsmanifest
CMake Error at cmake/Modules/FindLLVM.cmake:50 (message):
  Failed to execute llvm-config
  ('/nix/store/9g4chijh9255y65vf3nsgg9258knli58-llvm-5.0.2/bin/llvm-config',
  result code: '1)'
Call Stack (most recent call first):
  cmake/Modules/FindLLVM.cmake:97 (_LLVM_FAIL)
  cmake/Modules/FindLLVM.cmake:122 (llvm_set_libs)
  CMakeLists.txt:34 (find_package)


-- Configuring incomplete, errors occurred!
See also "/private/var/folders/d5/5ybzst6d2wbggcfhkjwl56s80000gn/T/nix-build-ldc-1.24.0.drv-0/ldc-1.24.0-src/build/CMakeFiles/CMakeOutput.log".
builder for '/nix/store/5k9zp5dzbw8s09qlv4a5jlv34s3hz0zh-ldc-1.24.0.drv' failed with exit code 1
error: build of '/nix/store/5k9zp5dzbw8s09qlv4a5jlv34s3hz0zh-ldc-1.24.0.drv' failed
```

**Comment by fabianhjr** [comment]
Hi @lionello, could you try removing the LLVM 5 version for macos and check if it works with LLVM 11?

**Comment by lionello** [comment]
> Hi @lionello, could you try removing the LLVM 5 version for macos and check if it works with LLVM 11?

That worked! Good call.

**Comment by fabianhjr** [comment]
Thanks @lionello , then I will remove the language split between linux and darwin.

**Comment by fabianhjr** [comment]
Ok, updated MacOS to LLVM 11

File: pkgs/development/compilers/ldc/generic.nix [hunk 1]
```diff
@@ -56,16 +56,14 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [
-    cmake ninja makeWrapper unzip ldcBootstrap lit lit.python
+    cmake ldcBootstrap lit lit.python llvm_11 makeWrapper ninja unzip
   ]
   ++ stdenv.lib.optionals stdenv.hostPlatform.isDarwin [
     darwin.apple_sdk.frameworks.Foundation
     # https://github.com/NixOS/nixpkgs/issues/57120
     # https://github.com/NixOS/nixpkgs/pull/59197#issuecomment-481972515
```
Comment (lionello) [inline pkgs/development/compilers/ldc/generic.nix:64]: What do these comments refer to now? Could probably remove these.

Comment (fabianhjr) [reply pkgs/development/compilers/ldc/generic.nix:64]: Don't think so, those are build dependencies and I didn't saw dependency changes on the release notes of LDC.

Comment (lionello) [reply pkgs/development/compilers/ldc/generic.nix:64]: Those were comments for `llvm_5` so now that `llvm_5` got removed, I think we can remove the comments as well.

Comment (fabianhjr) [reply pkgs/development/compilers/ldc/generic.nix]: Resolved, sorry got confused by the diff of the nativeBuildInputs.

**Review by lionello — APPROVED** [review]

**Review by lionello — APPROVED** [review]

**Comment by lionello** [comment]
@worldofpeace could you help merge this?

**Pull request merged**